File: design-patterns/go/composite/transport_list/transport_group.go
// transport_group.go

package main

type TransportGroup struct {
	transportList []Transport
}

func NewTransportGroup() (transportGroup *TransportGroup) {
	transportGroup = &TransportGroup{}
	// transportGroup.transportList = NewArrayList()
	return
}

func (transportGroup *TransportGroup) AddTransport(transport Transport) {
	transportGroup.transportList = append(transportGroup.transportList, transport)
}

func (transportGroup *TransportGroup) Operate() {
	for _, transport := range transportGroup.transportList {
		transport.Operate()
	}
}

func (transportGroup *TransportGroup) RemoveTransport(transport Transport) {
	newTransportGroup := []Transport{}

	for _, currentTransport := range transportGroup.transportList {
		if currentTransport == transport {
			continue
		}

		newTransportGroup = append(newTransportGroup, currentTransport)
	}

	transportGroup.transportList = newTransportGroup
}

func (transportGroup *TransportGroup) Start() {
	for _, transport := range transportGroup.transportList {
		transport.Start()
	}
}

func (transportGroup *TransportGroup) Stop() {
	for _, transport := range transportGroup.transportList {
		transport.Stop()
	}
}

File: design-patterns/ts/composite/transport-list/transport-group.ts
// transport-group.ts

import Transport from "./transport";

class TransportGroup implements Transport {
    private transportList: Transport[] = [];

    start(): void {
        for (let transport of this.transportList) {
            transport.start();
        }
    }

    operate(): void {
        for (let transport of this.transportList) {
            transport.operate();
        }
    }

    stop(): void {
        for (let transport of this.transportList) {
            transport.stop();
        }
    }

    addTransport(transport: Transport): void {
        this.transportList.push(transport);
    }

    removeTransport(transport: Transport): void {
        this.transportList.splice(this.transportList.indexOf(transport), 1);
    }
}

export default TransportGroup;